core/src/kmx/kmx_xstring.h

```diff
@@ -149,6 +171,50 @@ u16string_to_u32string(const std::u16string &source) {
   return out;
 }
 
+inline bool Uni_IsEndOfPlaneNonCharacter(km_kbp_usv ch) {
+  return (((ch) & Uni_FFFE_NONCHARACTER) == Uni_FFFE_NONCHARACTER); // matches FFFF or FFFE
+}
+
+inline bool Uni_IsNoncharacter(km_kbp_usv ch) {
+  return (((ch) >= Uni_FD_NONCHARACTER_START && (ch) <= Uni_FD_NONCHARACTER_END) || Uni_IsEndOfPlaneNonCharacter(ch));
+}
+
+inline bool Uni_InCodespace(km_kbp_usv ch) {
+  return ((ch) <= 0x10FFFF);
```

Comment:
Can we get a const for this as well given we have all the others?

Reply:
done